### ISSUE TITLE
Support deepObject style for query params

### ISSUE THREAD

**Issue by czechboy0** [apple/swift-openapi-generator#259]
Support deepObject style for query params.

Used by the Stripe API, for example.

**Comment by czechboy0** [comment]
Not just for query items, but also URL encoded bodies, once we support them.

**Comment by tib** [comment]
I just encountered this issue and I'd love to see better support for query params, let me know if you need help with it. :)

**Comment by czechboy0** [comment]
Yes support for at least deepObject is planned for 1.0, and it would be awesome if you could contribute it. I suspect most of the changes would be in the runtime library's URIEncoder and URIDecoder, please take a look there and see if adding deepObject support would be a project you want to take on. Note that while it's being improved, it might also make sense to just do https://github.com/apple/swift-openapi-generator/issues/258 as well, as the latter might be a natural extension of the existing logic. deepObject will be slightly more involved, but shouldn't be too difficult.

Take a look at how the encoder/decoder works today and see if you can extend it, and I'm here if you have any questions 🙂

**Comment by czechboy0** [comment]
I'll assign you to the issue for now, but don't feel obligated if you decide not to do it, just let me know and I'll unassign it. It's to avoid multiple people working on the same issue 🙂

**Comment by czechboy0** [comment]
A related ask for this: https://github.com/apple/swift-openapi-generator/issues/292

**Comment by czechboy0** [comment]
Ok just let us know if you start working on this @tib, but no pressure. I unassigned it for now to leave it available for others to pick up.

**Comment by kstefanou52** [comment]
Hi @czechboy0, I encountered the same issue so I took the initiative to address it. 
Excited to hear your thoughts when you get a chance to check my PRs out. Cheers! 🚀

- https://github.com/apple/swift-openapi-runtime/pull/100
- https://github.com/apple/swift-openapi-generator/pull/538

**Comment by czechboy0** [comment]
Runtime changes landed in swift-openapi-runtime 1.4.0.

**Comment by czechboy0** [comment]
Landed in #538
